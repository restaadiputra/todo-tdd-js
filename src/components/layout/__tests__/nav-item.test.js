import React from 'react';
import NavItem from '../nav-item';
import { renderWithRouter } from 'test/utils';

test('render without crash', () => {
  const { getByText, container } = renderWithRouter(
    <NavItem path="/">Link</NavItem>
  );

  expect(getByText('Link')).toBeInTheDocument();
  expect(container.firstChild.getAttribute('href')).toBe('/');
  expect(container.firstChild).toMatchSnapshot();
});

test('render without label', () => {
  const { container } = renderWithRouter(<NavItem path="/"></NavItem>);
  expect(container.firstChild.firstChild.textContent).toBe('');
});

test('render without active css if route is not match', () => {
  const { container } = renderWithRouter(<NavItem path="/testing"></NavItem>);
  expect(container.firstChild).toMatchSnapshot();
});
