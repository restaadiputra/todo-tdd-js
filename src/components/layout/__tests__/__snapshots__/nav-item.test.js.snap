// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render without crash 1`] = `
<a
  aria-current="page"
  class="w-full px-3 py-2 rounded text-gray-400 hover:bg-gray-900 hover:text-white active"
  href="/"
>
  <span>
    Link
  </span>
</a>
`;
